packages/order-utils/src/signature_utils.ts

```diff
@@ -278,7 +286,50 @@ export const signatureUtils = {
         } catch (err) {
             // Detect if Metamask to transition users to the MetamaskSubprovider
             if ((provider as any).isMetaMask) {
-                throw new Error(OrderError.InvalidMetamaskSigner);
+                throw new Error(TypedDataError.InvalidMetamaskSigner);
+            } else {
+                throw err;
+            }
+        }
+    },
+    /**
+     * Signs an order using `eth_signTypedData` and returns a SignedOrder.
+     * @param   supportedProvider      Web3 provider to use for all JSON RPC requests
+     * @param   transaction The ZeroEx Transaction to sign.
+     * @param   signerAddress   The hex encoded Ethereum address you wish to sign it with. This address
+     *          must be available via the supplied Provider.
+     * @return  A SignedOrder containing the order and Elliptic curve signature with Signature Type.
+     */
+    async ecSignTypedDataTransactionAsync(
```

Comment:
Let's also add a CHANGELOG entry for `order-utils` with a minor version bump detailing the addition of this method.

Reply:
shouldn't it be major version bump because we also change `OrderError` to `TypedDataError`?